### ISSUE TITLE
JPEG: JHOVE skips wrong number of bytes in unknown APP0 segments

### ISSUE THREAD

**Issue by marhop** [openpreserve/jhove#302]
Hello,

when validating some JPEG files that contain APP0 segments with the identifier string "Adobe_CM" JHOVE 1.18.1 raises the error "Expected marker byte 255, got 2". While APP0 segments usually have the identifier "JFIF" or "JFXX" in JPEG/JFIF files, application specific identifier strings are not illegal per the specifications:

> Additional APP0 marker segments can be used to hold application-specific information which does not affect the decodability or displayability of the JFIF file. Application specific APP0 marker segments must appear after the JFIF APP0 and any JFXX APP0 segments. Decoders should skip any unrecognized application-specific APP0 segments. ([JFIF 1.02](http://www.w3.org/Graphics/JPEG/jfif3.pdf), p. 3)

> APPn other than APP1 and APP2 or COM segments are not used by Exif. However Exif readers should be designed skip over unknown APPn and COM. ([Exif 2.1](http://www.exif.org/dcf-exif.PDF), p. 14/[Exif 2.2](http://www.exif.org/Exif2-2.PDF), p. 11)

Consequently, JHOVE simply skips unknown APP0 segments with an identifier other than "JFIF" or "JFXX". However, it skips the wrong amount of bytes sometimes and thus does not jump to the beginning of the next segment (where a 0xFF/255 byte would be expected).

Long story short, I will send you a pull request! (#303)

Regards,
Martin
